src/pages/index.js

```diff
@@ -105,36 +138,75 @@ function handleAddNewCardButton() {
 }
 
 function handleProfileFormSubmit(inputValues) {
+  document.querySelector(settings.submitButtonSelector).textContent =
+    "Saving...";
+  api
+    .editUserInfo(inputValues)
+    .then((data) => {
+      // process the result
+      const name = data.name;
+      const subtitle = data.about;
+      return { name, subtitle };
+    })
+    .finally(() => {
+      document.querySelector(settings.submitButtonSelector).textContent =
+        "Save";
+    });
   userInfo.setUserInfo(inputValues);
   addProfilePopup.closeModal();
```

Comment:
```js

  userInfo.setUserInfo(inputValues);
  addProfilePopup.closeModal();
```


This should be inside block `then`

You need to update `DOM` only in blocks `then` after a successful response from the server so that when there is an error it wouldn’t change